Mathlib/LinearAlgebra/AffineSpace/AffineSubspace/Basic.lean

```diff
@@ -877,4 +877,42 @@ lemma affineSpan_pair_eq_of_right_mem_of_ne {p₁ p₂ p₃ : P} (h : p₁ ∈ l
     line[k, p₂, p₁] = line[k, p₂, p₃] :=
   affineSpan_pair_eq_of_mem_of_mem_of_ne (left_mem_affineSpan_pair _ _ _) h hne.symm
 
+/-- Given two triples of points, if the lines determined by corresponding pairs of points are
+parallel, or more generally the directions of the lines in the second triple are contained in those
+of the lines in the first triple with one pair being parallel (this does not provide extra
+mathematical generality, but can sometimes be more convenient when using this lemma), and one of
+the points in the first triple does not lie on the line through the other two (typically, if they
+are not collinear), then the vectors between corresponding pairs of points are all related by the
+same nonzero scale factor. -/
```

Comment:
It's great that a doc string is provided (we should add far more) I find this text very hard to parse. I think one way to improve it would be to make it a lot less precise: after all, the formal statement exists.